### PR TITLE
Add config to skip schedules of tables with TWCS

### LINKED ISSUE #151: Don't repair TWCS tables by default
We should add a flag to disable repairs of TWCS tables by default

### PR DESCRIPTION
**Pull request by masokol**
The default is set to false, i.e same behavior as before this change. On top of that even if user sets the option to true, i.e to skip repair of TWCS tables this can be overriden in the schedule.yaml for specific tables.
Closes #151

### CODE REVIEW

File: application/src/main/resources/ecc.yml [hunk 1]
```diff
@@ -79,6 +79,12 @@ repair:
   ## The default FileBasedRepairConfiguration uses a schedule.yml file to define per-table configurations.
   ##
   provider: com.ericsson.bss.cassandra.ecchronos.application.FileBasedRepairConfiguration
+
+  ##
+  ## Specifies if tables with TWCS (TimeWindowCompactionStrategy) should be ignored for repair
+  ##
+  ignore_twcs_tables: false
```
Comment (itskarlsson) [inline application/src/main/resources/ecc.yml:86]: Nit: Have this setting after lock_type to keep the most integral settings towards the top

File: core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/DefaultRepairConfigurationProvider.java [hunk 2]
```diff
@@ -70,22 +76,22 @@ public void onKeyspaceChanged(KeyspaceMetadata current, KeyspaceMetadata previou
     }
 
     @Override
-    public void onTableAdded(TableMetadata table)
+    public void onTableAdded(TableMetadata tableMetadata)
```
Comment (itskarlsson) [inline core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/DefaultRepairConfigurationProvider.java:79]: Nit: Is this change necessary? table is clear and easily readable.

File: core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/DefaultRepairConfigurationProvider.java [hunk 3]
```diff
@@ -105,11 +111,29 @@ private void allTableOperation(String keyspaceName, Consumer<TableReference> con
...
+            if (!shouldIgnoreTable(tableMetadata))
+            {
+                TableReference tableReference = myTableReferenceFactory.forTable(keyspaceName, tableMetadata.getName());
+                consumer.accept(tableReference);
+            }
+        }
+    }
 
-            consumer.accept(tableReference);
+    private boolean shouldIgnoreTable(TableMetadata tableMetadata)
```
Comment (itskarlsson) [inline core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/DefaultRepairConfigurationProvider.java:122]: isTableIgnored might be better

**Review by tommystendahl — APPROVED** [review]

File: application/src/main/resources/ecc.yml [hunk 4]
```diff
@@ -79,6 +79,7 @@ repair:
   ## The default FileBasedRepairConfiguration uses a schedule.yml file to define per-table configurations.
   ##
   provider: com.ericsson.bss.cassandra.ecchronos.application.FileBasedRepairConfiguration
+
```
Comment (itskarlsson) [inline application/src/main/resources/ecc.yml:82]: Remove this newline

File: application/src/main/resources/ecc.yml [hunk 5]
```diff
@@ -95,6 +96,10 @@ repair:
   ##
   lock_type: vnode
   ##
+  ## Specifies if tables with TWCS (TimeWindowCompactionStrategy) should be ignored for repair
+  ##
+  ignore_twcs_tables: false
```
Comment (itskarlsson) [inline application/src/main/resources/ecc.yml:101]: NIT: Add the parameter towards the end of the file as it is far more niche than alarms and intervals

File: core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/DefaultRepairConfigurationProvider.java [hunk 6]
```diff
@@ -72,7 +78,7 @@ public void onKeyspaceChanged(KeyspaceMetadata current, KeyspaceMetadata previou
     @Override
     public void onTableAdded(TableMetadata table)
     {
-        if (myReplicatedTableProvider.accept(table.getKeyspace().getName()))
+        if (myReplicatedTableProvider.accept(table.getKeyspace().getName()) && !isTableIgnored(table))
```
Comment (itskarlsson) [inline core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/DefaultRepairConfigurationProvider.java:81]: Would this logic not be better suited in ReplicatedTableProvider? We already have logic which ignores tables in ReplicatedTableProvider.accept() and it would be nice to have all ignore logic in the same place.

**Review by itskarlsson — CHANGES_REQUESTED** [review]

**Review by tommystendahl — APPROVED** [review]

File: core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/DefaultRepairConfigurationProvider.java (hunk 6 above)
Comment (masokol) [reply core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/DefaultRepairConfigurationProvider.java]: I don't think so, replicatedTableProvider.accept is used for both ondemand and schedules. This setting is global and can be overriden in schedule.yaml for the specific tables. I've moved the code around a bit so the validation is done in the same place as validation if it's disabled.

**Review by itskarlsson — APPROVED** [review]
